stdlib/public/core/Integers.swift

```diff
@@ -1315,6 +1315,152 @@ extension BinaryInteger {
     return self.magnitude % other.magnitude == 0
   }
 
+  /// Returns `true` if this value is a power of the given base, and `false`
+  /// otherwise.
+  ///
+  /// For two integers *a* and *b*, *a* is a power of *b* if *a* is equal to
+  /// any repeated multiplication of *b*. For example, *16* is a power of *2*
+  /// because _16 = 2 * 2 * 2 * 2_. *-27* is a power of *-3* because
+  /// _-27 = -3 * -3 * -3_. *1* is a power of everything because any number to
+  /// the zero power is one.
```

Comment:
The examples might be more readable in a list, using code spans instead of emphasis.

```suggestion
  /// For two integers *a* and *b*, *a* is a power of *b* if *a* is equal to
  /// any repeated multiplication of *b*. For example:
  ///
  /// * `16.isPower(of: 2)` because `16 == (2 * 2 * 2 * 2)`
  /// * `(-27).isPower(of: -3)` because `-27 == (-3 * -3 * -3)`
  ///
  /// One is a power of everything because any number to the zero power is one.
```

Reply:
> /// One is a power of everything because any number to the zero power is one.

This is not exactly right; one is a power of everything because the empty product is one because one is the multiplicative identity. "Any number to the zero power is one" also follows from the convention for the empty product. I'm not sure how far down this rabbit hole the comment actually needs to go, however =)

Reply:
Yes, this is more precise from a scientific perspective. I have improved the comments by adding `empty product`.